### PR TITLE
48 ta ms create match button logic

### PR DESCRIPTION
**Pull request by MayaS1111**
- Matches mentee with mentor once match button is pressed on the mentor's profile page
     - Button only displays if mentee is in the same cohort as mentor
     - Button only displays if mentor's capacity has not been reached
     - Button only displays if mentee is not currently matched with any mentor
- Mentee dashboard view:
     - Only shows mentors that have not reached their capacity cap
     - Says there are no more available mentors if all mentors have reached capacity
     - Says matching is closed if cohort is no longer active 
- New methods in User model:
     - name
     - matched? 
     - cohort
     - role
     - capacity
     - mentee_capacity_count(cohort_id)
- Nav routes have been added
- More changes to cohort_members association-related issues in Users and in the registration process that were not caught 
before
- Profile and Dashboard have a temporary section for current user info (Fadi will style it)
- Sample data has been updated:
     - Now uses cohort_members
     - Roles are lower cased to reduce confusion and solve issues with rendering

### CODE REVIEW

File: app/views/dashboard/_profile_card.html.erb
Comment (brvarner) [inline app/views/dashboard/_profile_card.html.erb:1]: I love the concision of the name method

**Review by brvarner — APPROVED** [review]
There is a lot of good stuff here, great work! I'll get more info from you later, everything looks good.

File: app/models/user.rb
Comment (rezahusain) [inline app/models/user.rb:1]: I like all the helper methods you created within the User model, will make it a lot more convenient to grab specific information moving forward.

File: app/views/shared/_navbar.html.erb [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <div>
     <ul class="flex space-x-14 text-base text-black">
       <li><a href="/" class="hover:text-gray-600">Home</a></li>
-      <li><a href="#" class="hover:text-gray-600">Mentor</a></li>
+      <li><a href="/dashboard/mentee" class="hover:text-gray-600">Mentor</a></li>
```
Comment (rezahusain) [inline app/views/shared/_navbar.html.erb:8]: I think the name "Mentor" in this situation should only be shown when the current user (mentee) is matched.

**Review by rezahusain — APPROVED** [review]
Looks solid overall! I'll approve.

**Pull request merged**